### PR TITLE
python310Packages.pixcat: init at 0.1.4

### PR DESCRIPTION
**Pull request by TilCreator**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by papojari** [comment]
If this is `python310Packages.pixcat` then why is `python39Packages.pixcat` being built too?

**Comment by papojari** [comment]
Running the binary yields:
```
[nix-shell:~/.cache/nixpkgs-review/pr-162556]$ ./results/python310Packages.pixcat/bin/pixcat 
Traceback (most recent call last):
  File "/nix/store/31v993wmnbx224pp7a52mfqwqy1cznr2-python3.10-pixcat-0.1.4/bin/.pixcat-wrapped", line 6, in <module>
    from pixcat.cli import main
  File "/nix/store/aswlxm65csr4zg6vpc2nl2dcs90japyg-python3.9-pixcat-0.1.4/lib/python3.9/site-packages/pixcat/__init__.py", line 6, in <module>
    from .image import Image
  File "/nix/store/aswlxm65csr4zg6vpc2nl2dcs90japyg-python3.9-pixcat-0.1.4/lib/python3.9/site-packages/pixcat/image.py", line 13, in <module>
    from PIL import Image as PILImage
  File "/nix/store/di5wrndvwlg3yvw7y264mybx7ab4gy1d-python3.9-Pillow-9.0.0/lib/python3.9/site-packages/PIL/Image.py", line 89, in <module>
    from . import _imaging as core
ImportError: cannot import name '_imaging' from 'PIL' (/nix/store/di5wrndvwlg3yvw7y264mybx7ab4gy1d-python3.9-Pillow-9.0.0/lib/python3.9/site-packages/PIL/__init__.py)
```
Are you sure that you tested all binaries like stated in your comment?

**Comment by papojari** [comment]
> If this is `python310Packages.pixcat` then why is `python39Packages.pixcat` being built too?

While `nixpkgs-review` says that both package were built I can only find the 3.10 binary.

**Comment by TilCreator** [comment]
I myself am a bit confused, I used `nix build .#python310Packages.pixcat` and `nix build .#python39Packages.pixcat` to build and test the binaries. Also the package should work with any Python version above 3.6, the name of the commit is not chosen very well.

**Comment by Artturin** [comment]
Python libraries are built for all supported python versions

Since pixcat is a cli tool as well as a library it should also be in `all-packages.nix`

`pixcat = with python3Packages; toPythonApplication pixcat;`

**Comment by Lassulus** [comment]
```
[nix-shell:~/.cache/nixpkgs-review/pr-162556]$ pixcat
Traceback (most recent call last):
  File "/nix/store/94slby15269b025idv3zyc3ndkrr372m-python3.9-pixcat-0.1.4/bin/.pixcat-wrapped", line 6, in <module>
    from pixcat.cli import main
  File "/nix/store/mns7r8m3gfgjaiiraxf7sprfychg54dj-python3.10-pixcat-0.1.4/lib/python3.10/site-packages/pixcat/__init__.py", line 6, in <module>
    from .image import Image
  File "/nix/store/mns7r8m3gfgjaiiraxf7sprfychg54dj-python3.10-pixcat-0.1.4/lib/python3.10/site-packages/pixcat/image.py", line 13, in <module>
    from PIL import Image as PILImage
  File "/nix/store/8byylwabglrkml3g3bmcvjyi10c4zvgr-python3.10-Pillow-9.0.1/lib/python3.10/site-packages/PIL/Image.py", line 89, in <module>
    from . import _imaging as core
ImportError: cannot import name '_imaging' from 'PIL' (/nix/store/8byylwabglrkml3g3bmcvjyi10c4zvgr-python3.10-Pillow-9.0.1/lib/python3.10/site-packages/PIL/__init__.py)
```
sadly not working for me

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6096,6 +6096,8 @@ in {
 
   pdunehd = callPackage ../development/python-modules/pdunehd { };
 
+  pixcat = callPackage ../development/python-modules/pixcat { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:6099]: If the package is not used as a python depdency it shouldn't be added here.

File: pkgs/development/python-modules/pixcat/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "657c8fe04513caecccd6086b347aa4b85db6b4c0f761b162cb9cd789abe7abb6";
+  };
+
+  propagatedBuildInputs = [
+    blessed
+    docopt
+    pillow
+    requests
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pixcat/default.nix:29]: Please add a pythonImportsCheck

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (TilCreator) [reply pkgs/top-level/python-packages.nix:6099]: The package is meant as a python dependency as well, is that enough?

**Pull request merged**